src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -31,6 +32,7 @@ protected BinaryWriter()
             OutStream = Stream.Null;
             _encoding = Encoding.UTF8;
             _useFastUtf8 = true;
+            _isDerivedWriter = GetType() != typeof(BinaryWriter);
```

Comment:
Note: You do not need to save the result into a field. `GetType() != typeof(X)` compiles into a single instruction.

Reply:
Ah, alright. As I mentioned, feel free to adjust it. Or let me know if you want me to change it.